cheatsheets/Protect_FileUpload_Against_Malicious_File.md

```diff
@@ -1,597 +1,138 @@
+- **Whitelist allowed extensions. Only allow safe and critical extensions for business functionality**
+  - **Ensure that [input validation](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Input_Validation_Cheat_Sheet.md#validating-free-form-unicode-text) is applied before validating the extensions.**
+- **Allowed users only can upload files**
+- **Validate the file type, don't trust the [Content-Type header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)  as it can be spoofed**
+- **Change the file name to something generated by the application**
+- **Set a file name length limit. Restrict the allowed characters if possible**
+  - **Setting a character limit could deny the attacker from sending null bytes, adding secondary extensions, and performing XSS attacks if the file name is reflected**
+- **Set a file size limit**
+- **Store the files on a different server. If that's not possible, store them in a private directory**
+  - **In the case of public access to the files, ensure that proper permissions are set, file names are safe, and encoded**
+- **Run the file through an antivirus or a sandbox if available to validate that it doesn't contain malicious or bogus data**
+- **Ensure the usage of secure and updated libraries for your needs**
 
-Usually, when an application expect his user to upload a document, the application expect to receive a document for which the intended use will be for reading/printing/archiving. The document should not alter is content at opening time and should be in a final rendered state.
+## Contents
 
-The most common file types used to transmit *malicious code* into file upload feature are the following:
+- [File Upload Threats](#file-upload-threats)
+    - [Malware Upload](#malware-upload)
+    - [Public File Retrieval](#public-file-retrieval)
+- [File Upload Protection](#file-upload-protection)
+    - [Extension Validation](#extension-validation)
+        - [Whitelist Extensions](#whitelist-extensions)
+        - [Blacklist Extensions](#blacklist-extensions)
+    - [User Permissions](#user-permissions)
+    - [Content-Type Validation](#content-type-validation)
+    - [File Name Sanitization](#file-name-sanitization)
+    - [File Storage Location](#file-storage-location)
+    - [System Permissions](#system-permissions)
+    - [Upload and Download Limits](#upload-and-download-limits)
+- [Java Code Snippets](#java-code-snippets)
 
-- Microsoft Office document: Word/Excel/PowerPoint using [VBA Macro](https://en.wikipedia.org/wiki/Visual_Basic_for_Applications) and [OLE package](https://en.wikipedia.org/wiki/Object_Linking_and_Embedding).
-- Adobe PDF document: Insert malicious code as attachment.
-- Images: Malicious code embedded into the file or use of binary file with image file extension.
+## File Upload Threats
```

Comment:
Would be worth a mention that file upload can be vulnerable to CSRF just like any other form - less common to see people exploiting it, but not one you'd want to forget.